sandbox/modules/identity/src/test/java/org/opensearch/identity/BearerAuthTests.java

```diff
@@ -0,0 +1,160 @@
+        jwtClaims.put("sub", "testSubject");
+        String encodedToken = JwtVendor.createJwt(jwtClaims);
+        String headerBody = "Bearer " + encodedToken;
+
+        Request request = new Request("GET", "/_cluster/health");
+        RequestOptions options = RequestOptions.DEFAULT.toBuilder().addHeader("Authorization", headerBody).build();
+        request.setOptions(options);
+
+        Response response = getRestClient().performRequest(request);
+
+        String content = new String(response.getEntity().getContent().readAllBytes(), StandardCharsets.UTF_8);
+
+        assertEquals(RestStatus.OK.getStatus(), response.getStatusLine().getStatusCode());
+        assertThat(content, containsString("green"));
+
+    }
+
+    public void testClusterHealthWithExpiredBearerAuthenticationHeader() throws IOException {
+
+        Map<String, String> jwtClaims = new HashMap<>();
+        jwtClaims.put("sub", "testSubject");
+        String encodedToken = JwtVendorTestUtils.createExpiredJwt(jwtClaims);
+        String headerBody = "Bearer " + encodedToken;
+
+        Request request = new Request("GET", "/_cluster/health");
+        RequestOptions options = RequestOptions.DEFAULT.toBuilder().addHeader("Authorization", headerBody).build();
+        request.setOptions(options);
+        // Should be unauthorized since JWT is expired
+        try {
+            getRestClient().performRequest(request);
+        } catch (ResponseException e) {
+            MatcherAssert.assertThat(e.getResponse().getStatusLine().getStatusCode(), equalTo(401));
+        }
+    }
+
+    public void testClusterHealthWithInvalidBearerAuthenticationHeader() throws IOException {
+
+        Map<String, String> jwtClaims = new HashMap<>();
+        jwtClaims.put("sub", "testSubject");
+        String encodedToken = JwtVendorTestUtils.createInvalidJwt(jwtClaims);
+        String headerBody = "Bearer " + encodedToken;
+
+        Request request = new Request("GET", "/_cluster/health");
+        RequestOptions options = RequestOptions.DEFAULT.toBuilder().addHeader("Authorization", headerBody).build();
+        request.setOptions(options);
+        try {
+            getRestClient().performRequest(request);
+        } catch (ResponseException e) {
+            MatcherAssert.assertThat(e.getResponse().getStatusLine().getStatusCode(), equalTo(401));
+        }
+    }
+
+    public void testClusterHealthWithCorruptBearerAuthenticationHeader() throws IOException {
+
+        String headerBody = "Bearer NotAJWT";
+
+        Request request = new Request("GET", "/_cluster/health");
+        RequestOptions options = RequestOptions.DEFAULT.toBuilder().addHeader("Authorization", headerBody).build();
+        request.setOptions(options);
+        try {
+            getRestClient().performRequest(request);
+        } catch (ResponseException e) {
+            MatcherAssert.assertThat(e.getResponse().getStatusLine().getStatusCode(), equalTo(401));
+        }
+    }
+}
```

Comment:
@cwperks @scrawfor99 This test suite seems like a mix of Unit (lines 32-91) and Integration (lines 93-159) tests. Should we separate them out into two classes. Thoughts?

Reply:
Whatever people think is best seems fine with me.  I see why you would separate them by test type, but I also get why keeping all the tests related to Bearer Auth stuff together is nice. Whichever is preferred I can do.

Reply:
Upon further review, I think it is best to leave things as-is for now. The primary reason being the dependency handling between `authn` and `identity`. Right now, `identity` is dependent on `authn` but all of the `JwtVendor*` classes are in `identity`. In order to move the unit tests over to the `AuthenticationTokenHandlerTests` class we would need to make `authn` dependent on `identity` or move the `AuthenticationTokenHandlerTests`.
If we end up combining `authn` and `identity` then this will no longer be an issue but for the time being it seems best to merge as-is and revisit down the road.